src/observable/ScalarObservable.ts

```diff
@@ -80,8 +80,8 @@ proto.reduce = function <T, R>(project: (acc: R, x: T) => R, seed?: R): Observab
     return <any>this;
   }
   let result = tryCatch(project)(seed, this.value);
-  if (result === errorObject) {
-    return new ErrorObservable(errorObject.e);
+  if (result as any === errorObject) {
```

Comment:
or like `=> Function | errorObject` maybe? (just throwing out thoughts, not sure if it works)

for casting syntax, since I was living in C# prefer `as` in usual cases.. but in here would like to go with consistency in single style. Does not feel to introduce different styles while original style serves.